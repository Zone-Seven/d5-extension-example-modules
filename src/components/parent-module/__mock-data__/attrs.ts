import {ParentModuleAttrs} from '../types';

export const defaultAttributes: ParentModuleAttrs = {};

export const customIcon: ParentModuleAttrs = {
  icon: {
    decoration: {
      icon: {
        desktop: {
          value: {
            unicode: '&#x39;',
            type: 'divi',
            weight: '400',
          },
        },
      }
    }
  },
};

export const iconStyle: ParentModuleAttrs = {
  ...defaultAttributes,
  icon: {
    decoration: {
      icon: {
        desktop: {
          value: {
            unicode: '&#x39;',
            type: 'divi',
            weight: '400',
            color: '#ae16f0',
            size: '27px',
          },
        },
      }
    }
  },
};

export const titleStyle: ParentModuleAttrs = {
  title: {
    decoration: {
      font: {
        font: {
          desktop: {
            value: {
              color: '#ae16f0',
              weight: '300',
              style: [
                'capitalize',
              ],
            },
          },
        },
      }
    }
  },
};

export const contentStyle: ParentModuleAttrs = {
  content: {
    decoration: {
      bodyFont: {
        body: {
          font: {
            desktop: {
              value: {
                color: '#ae16f0',
                size: '16px',
                style: [
                  'uppercase',
                ],
              },
            },
          },
        },
      },
    }
  }
}
