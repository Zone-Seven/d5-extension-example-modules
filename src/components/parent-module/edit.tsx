// External Dependencies.
import React, { ReactElement } from 'react';

// Divi Dependencies.
import { ChildModulesContainer, ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { ParentModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { ModuleStyles } from './styles';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {ParentModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const ParentModuleEdit = (props: ParentModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
    childrenIds,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      tag="ul"
    >
      <ChildModulesContainer ids={childrenIds} parentId={id} />
    </ModuleContainer>
  );
}

export {
  ParentModuleEdit,
};
