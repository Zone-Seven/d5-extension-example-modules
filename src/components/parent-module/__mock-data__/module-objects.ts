import Immutable from 'seamless-immutable';

import { defaultAttributes, customIcon, iconStyle, titleStyle, contentStyle } from './attrs';
import {
  defaultAttributes as childDefaultAttributes,
} from '../../child-module/__mock-data__/attrs';
import { ExampleEditPostStoreState, ExampleMutableEditPostStoreState } from '../../../types';

// Mutable state. NOTE: Update visualization if you change the mutable state structure.
/**
 * State visualization:
 *
 * **** root ****************************************************************************
 * *                                                                                    *
 * *    **** section1 **************************************************************    *
 * *    *                                                                          *    *
 * *    *    **** row1 ********************************************************    *    *
 * *    *    *                                                                *    *    *
 * *    *    *    ** column1 *********************************************    *    *    *
 * *    *    *    *                                                      *    *    *    *
 * *    *    *    *    ** parentModule1 *****************************    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule1           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule2           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule3           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    **********************************************    *    *    *    *
 * *    *    *    *                                                      *    *    *    *
 * *    *    *    *    ** parentModule2 *****************************    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule4           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule5           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule6           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    **********************************************    *    *    *    *
 * *    *    *    *                                                      *    *    *    *
 * *    *    *    *    ** parentModule3 *****************************    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule7           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule8           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule9           *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    **********************************************    *    *    *    *
 * *    *    *    *                                                      *    *    *    *
 * *    *    *    *    ** parentModule4 *****************************    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule10          *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule11          *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule12          *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    **********************************************    *    *    *    *
 * *    *    *    *                                                      *    *    *    *
 * *    *    *    *    ** parentModule5 *****************************    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule13          *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule14          *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *    *           childModule15          *    *    *    *    *    *
 * *    *    *    *    *    ************************************    *    *    *    *    *
 * *    *    *    *    *                                            *    *    *    *    *
 * *    *    *    *    **********************************************    *    *    *    *
 * *    *    *    *                                                      *    *    *    *
 * *    *    *    ********************************************************    *    *    *
 * *    *    *                                                                *    *    *
 * *    *    ******************************************************************    *    *
 * *    *                                                                          *    *
 * *    ****************************************************************************    *
 * *                                                                                    *
 * **************************************************************************************.
 *
 */
export const mutableState: ExampleMutableEditPostStoreState = {
  content: {
    root: {
      id:       'root',
      name:     'divi/root',
      children: ['section1'],
      props:    {},
    },
    section1: {
      id:       'section1',
      name:     'divi/section',
      parent:   'root',
      children: ['row1'],
      props:    {},
    },
    row1: {
      id:       'row1',
      name:     'divi/row',
      parent:   'section1',
      children: ['column1'],
      props: {
        attrs: {
          module: {
            advanced: {
              columnStructure: {
                desktop: {
                  value: '4_4',
                },
              },
            },
          },
        },
      },
    },
    column1: {
      id:       'column1',
      name:     'divi/column',
      parent:   'row1',
      children: ['module1'],
      props: {
        attrs: {
          module: {
            advanced: {
              type: {
                desktop: {
                  value: '4_4',
                },
              },
            }
          }
        },
      },
    },
    parentModule1: {
      id:     'parentModule1',
      name:   'example/parent-module',
      parent: 'column1',
      props:  {
        attrs: defaultAttributes,
      },
      children: [
        'childModule1',
        'childModule2',
        'childModule3',
      ],
    },
    childModule1: {
      id:     'childModule1',
      name:   'example/child-module',
      parent: 'parentModule1',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule2: {
      id:     'childModule2',
      name:   'example/child-module',
      parent: 'parentModule1',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule3: {
      id:     'childModule3',
      name:   'example/child-module',
      parent: 'parentModule1',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    parentModule2: {
      id:     'parentModule2',
      name:   'example/parent-module',
      parent: 'column1',
      props:  {
        attrs: customIcon,
      },
      children: [
        'childModule4',
        'childModule5',
        'childModule6',
      ],
    },
    childModule4: {
      id:     'childModule4',
      name:   'example/child-module',
      parent: 'parentModule2',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule5: {
      id:     'childModule5',
      name:   'example/child-module',
      parent: 'parentModule2',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule6: {
      id:     'childModule6',
      name:   'example/child-module',
      parent: 'parentModule2',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    parentModule3: {
      id:     'parentModule3',
      name:   'example/parent-module',
      parent: 'column1',
      props:  {
        attrs: iconStyle,
      },
      children: [
        'childModule7',
        'childModule8',
        'childModule9',
      ],
    },
    childModule7: {
      id:     'childModule7',
      name:   'example/child-module',
      parent: 'parentModule3',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule8: {
      id:     'childModule8',
      name:   'example/child-module',
      parent: 'parentModule3',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule9: {
      id:     'childModule9',
      name:   'example/child-module',
      parent: 'parentModule3',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    parentModule4: {
      id:     'parentModule4',
      name:   'example/parent-module',
      parent: 'column1',
      props:  {
        attrs: titleStyle,
      },
      children: [
        'childModule10',
        'childModule11',
        'childModule12',
      ],
    },
    childModule10: {
      id:     'childModule10',
      name:   'example/child-module',
      parent: 'parentModule4',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule11: {
      id:     'childModule11',
      name:   'example/child-module',
      parent: 'parentModule4',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule12: {
      id:     'childModule12',
      name:   'example/child-module',
      parent: 'parentModule4',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    parentModule5: {
      id:     'parentModule5',
      name:   'example/parent-module',
      parent: 'column1',
      props:  {
        attrs: contentStyle,
      },
      children: [
        'childModule13',
        'childModule14',
        'childModule15',
      ],
    },
    childModule13: {
      id:     'childModule13',
      name:   'example/child-module',
      parent: 'parentModule5',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule14: {
      id:     'childModule14',
      name:   'example/child-module',
      parent: 'parentModule5',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
    childModule15: {
      id:     'childModule15',
      name:   'example/child-module',
      parent: 'parentModule5',
      props:  {
        attrs: childDefaultAttributes,
      },
      children: [],
    },
  },
};

export const state: ExampleEditPostStoreState = Immutable(mutableState);
