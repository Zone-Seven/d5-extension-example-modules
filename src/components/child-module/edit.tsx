// External Dependencies.
import React, { ReactElement } from 'react';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';
import { processFontIcon } from '@divi/icon-library';

// Local Dependencies.
import { ChildModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { defaultAttrs as parentDefaultAttrs } from '../parent-module/constants';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {ChildModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const ChildModuleEdit = (props: ChildModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
    parentAttrs,
  } = props;

  // Merge parent module default values with parent module attributes.
  const parentModuleAttrs = mergeAttrs({
    defaultAttrs: parentDefaultAttrs,
    attrs: parentAttrs,
  });

  // Merge module default values with module attributes.
  const childModuleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const moduleAttrs = {
    ...childModuleAttrs,
    icon: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.icon,
      attrs: childModuleAttrs?.icon,
    }),
    iconColor: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.iconColor,
      attrs: childModuleAttrs?.iconColor,
    }),
    iconSize: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.iconSize,
      attrs: childModuleAttrs?.iconSize,
    }),
    titleFont: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.titleFont,
      attrs: childModuleAttrs?.titleFont,
    }),
    text: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.text,
      attrs: childModuleAttrs?.text,
    }),
    bodyFont: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.bodyFont,
      attrs: childModuleAttrs?.bodyFont,
    }),
  };
  
  const icon    = moduleAttrs?.icon?.desktop?.value;
  const title   = moduleAttrs?.title?.desktop?.value;
  const content = moduleAttrs?.content?.desktop?.value;

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      tag="li"
    >
      { icon && (
        <div className="child-module__icon et-pb-icon">
            {processFontIcon(icon)}
        </div>
      )}
      <div className="child-module__content-container">
        <div className="child-module__title">{title}</div>
        <div
          className="child-module__content"
          dangerouslySetInnerHTML={{ __html: content }}
        />
      </div>
    </ModuleContainer>
  );
}

export {
  ChildModuleEdit,
};
