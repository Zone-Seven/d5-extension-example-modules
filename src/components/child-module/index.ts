// Divi dependencies.
import { ModuleRegisterDefinition } from '@divi/module-library';

// Local dependencies.
import metadata from './module.json';
import { ChildModuleEdit } from './edit';
import { SettingsContent } from './settings-content';
import { SettingsDesign } from './settings-design';
import { SettingsAdvanced } from './settings-advanced';
import { ChildModuleAttrs } from './types';
import { placeholderContent } from './placeholder-content';

// Styles.
import './style.scss';
import './module.scss';
import { ModuleMetadata } from '@divi/types';

/**
 * Call To Action module.
 *
 * @since ??
 */
export const childModule: ModuleRegisterDefinition<ChildModuleAttrs> = {
  // Imported json has no inferred type hence type-cast is necessary.
  metadata: metadata as ModuleMetadata,
  placeholderContent,
  settings:   {
    content:  SettingsContent,
    design:   SettingsDesign,
    advanced: SettingsAdvanced,
  },
  renderers: {
    edit: ChildModuleEdit,
  },
  parentsName: ['example/parent-module'],
};
