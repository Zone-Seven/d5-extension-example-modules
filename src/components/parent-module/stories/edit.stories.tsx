import {
  createElement,
  ReactElement,
} from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { Story } from '@storybook/react';
import {
  defaultAttributes,
} from '../__mock-data__/attrs';

import { ParentModuleEdit } from '../edit';
import { ParentModuleEditProps } from '../types';
import { dispatch, select } from '@divi/data';
import { registerEditPostStore } from '@divi/edit-post';
import { parentModule } from '..';
import { childModule } from '../../child-module';
import { state } from '../__mock-data__/module-objects';
import { isNull } from 'lodash';
import { registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerModalLibraryStore, registerModals } from '@divi/modal-library';
import { registerAppUiStore } from '@divi/app-ui';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { registerSettingsStore } from '@divi/settings';
import { registerHistoryStore } from '@divi/history';
import { registerAppPreferencesStore } from '@divi/app-preferences';

// Create template to render argument given.
const templateDividerEdit: Story<ParentModuleEditProps> = args => createElement(ParentModuleEdit, args);

const name = 'divi/parent-module';

// Stories.
const DefaultValues      = templateDividerEdit.bind({}) as Story<ParentModuleEditProps>;
DefaultValues.args       = {
  name,
  id:    'parent-module-1',
  attrs: defaultAttributes,
  childrenIds: [
    'childModule1',
    'childModule2',
    'childModule3',
  ],
};
DefaultValues.parameters = {
  docs: {
    description: {
      story: 'Default attributes that gets added when Child module is added.',
    },
  },
};


export {
  DefaultValues,
};

export default {
  title: 'Modules/Parent Module/Edit',
  decorators: [
    ((story: () => ReactElement): ReactElement => {
      registerAppPreferencesStore();
      registerAppUiStore();
      registerEventsStore();
      registerKeyboardShortcutsStore();
      registerModals();
      registerSettingsStore();
      registerSerializedPostStore();
      registerEditPostStore(state);
      registerHistoryStore();
      if (isNull(select('divi/module-library'))) {
        registerModuleLibraryStore();
      }

      dispatch('divi/module-library').addModule(childModule);
      dispatch('divi/module-library').addModule(parentModule);
      return story();
    }),
  ],
};
