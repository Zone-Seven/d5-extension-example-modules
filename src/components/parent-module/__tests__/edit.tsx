import { createElement } from 'react';
import { shallow } from 'enzyme';
import {
  forEach,
  pickBy,
  has,
  get,
} from 'lodash';

import { registerAppUiStore } from '@divi/app-ui';
import { registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerModals } from '@divi/modal-library';
import { registerEditPostStore } from '@divi/edit-post';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { ParentModuleEdit } from '../edit';
import { dispatch } from '@divi/data';
import { parentModule } from '..';

import * as editStories  from '../stories/edit.stories';
import { state } from '../__mock-data__/module-objects';
import { childModule } from '../../child-module';
import { registerAppPreferencesStore } from '@divi/app-preferences';

// Only test story that has description. This naturally excludes `default` export from being included as test.
const testableStories = pickBy(editStories, story => has(story, 'parameters.docs.description.story')) as Omit<typeof editStories, 'default'>;

beforeAll(() => {
  registerModuleLibraryStore();
  dispatch('divi/module-library').addModule(childModule);
  dispatch('divi/module-library').addModule(parentModule);
});

beforeEach(() => {
  registerAppPreferencesStore();
  registerAppUiStore();
  registerEventsStore();
  registerKeyboardShortcutsStore();
  registerModals();
  registerSerializedPostStore();
  registerEditPostStore(state);
});

describe('<ParentModuleEdit />', () => {
  // Snapshot test based on storybook configuration.
  // Prevents regression due to unintentionally modifying component.
  forEach(testableStories, testableStory => {
    it(`Should match the snapshot when "${get(testableStory, 'parameters.docs.description.story') as string}"`, () => {
      expect(
        shallow(createElement(ParentModuleEdit, testableStory.args)).html(),
      ).toMatchSnapshot();
    });
  });
});
