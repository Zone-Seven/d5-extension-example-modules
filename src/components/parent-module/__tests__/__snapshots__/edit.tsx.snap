// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParentModuleEdit /> Should match the snapshot when "Default attributes that gets added when Parent module is added." 1`] = `"<ul class=\\"parent-module_parentModule1 parent-module et_pb_module et-animated--vb\\" data-id=\\"parentModule1\\"><li class=\\"child-module_childModule1 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule1\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></li><li class=\\"child-module_childModule2 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule2\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></li><li class=\\"child-module_childModule3 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule3\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></li></ul>"`;

exports[`<ParentModuleEdit /> Should match the snapshot when "Display content style properly when the content style is set." 1`] = `"<ul class=\\"parent-module_parentModule5 parent-module et_pb_module et-animated--vb\\" data-id=\\"parentModule5\\"><li class=\\"child-module_childModule13 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule13\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule13 .et_pb_child_module_content {text-transform: uppercase; color: #ae16f0; font-size: 16px;}</style></li><li class=\\"child-module_childModule14 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule14\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule14 .et_pb_child_module_content {text-transform: uppercase; color: #ae16f0; font-size: 16px;}</style></li><li class=\\"child-module_childModule15 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule15\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule15 .et_pb_child_module_content {text-transform: uppercase; color: #ae16f0; font-size: 16px;}</style></li></ul>"`;

exports[`<ParentModuleEdit /> Should match the snapshot when "Display icon style properly when the icon style is set." 1`] = `"<ul class=\\"parent-module_parentModule3 parent-module et_pb_module et-animated--vb\\" data-id=\\"parentModule3\\"><li class=\\"child-module_childModule7 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule7\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule7 .et_pb_child_module_icon.et-pb-icon {color: #ae16f0;}.child-module_childModule7 .et_pb_child_module_icon.et-pb-icon {font-size: 27px;}</style></li><li class=\\"child-module_childModule8 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule8\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule8 .et_pb_child_module_icon.et-pb-icon {color: #ae16f0;}.child-module_childModule8 .et_pb_child_module_icon.et-pb-icon {font-size: 27px;}</style></li><li class=\\"child-module_childModule9 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule9\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule9 .et_pb_child_module_icon.et-pb-icon {color: #ae16f0;}.child-module_childModule9 .et_pb_child_module_icon.et-pb-icon {font-size: 27px;}</style></li></ul>"`;

exports[`<ParentModuleEdit /> Should match the snapshot when "Display selected icon properly into child module when the icon value is set." 1`] = `"<ul class=\\"parent-module_parentModule2 parent-module et_pb_module et-animated--vb\\" data-id=\\"parentModule2\\"><li class=\\"child-module_childModule4 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule4\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></li><li class=\\"child-module_childModule5 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule5\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></li><li class=\\"child-module_childModule6 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule6\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></li></ul>"`;

exports[`<ParentModuleEdit /> Should match the snapshot when "Display title style properly when the title style is set." 1`] = `"<ul class=\\"parent-module_parentModule4 parent-module et_pb_module et-animated--vb\\" data-id=\\"parentModule4\\"><li class=\\"child-module_childModule10 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule10\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule10 .et_pb_child_module_title {font-weight: 300; font-variant: small-caps; color: #ae16f0;}</style></li><li class=\\"child-module_childModule11 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule11\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule11 .et_pb_child_module_title {font-weight: 300; font-variant: small-caps; color: #ae16f0;}</style></li><li class=\\"child-module_childModule12 child-module et_pb_module et-animated--vb\\" data-id=\\"childModule12\\"><div class=\\"child-module__icon et-pb-icon\\">N</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.child-module_childModule12 .et_pb_child_module_title {font-weight: 300; font-variant: small-caps; color: #ae16f0;}</style></li></ul>"`;
